feat: add unit tests for AddMember.tsx

ashutoshsao opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
Feature (New Unit Tests) & Refactoring (Internationalization Fixes)
<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

<!--Add related issue number here.-->
Fixes #5963

**Snapshots/Videos:**
<img width="768" height="256" alt="screenshot-2025-12-31_12-55-02" src="https://github.com/user-attachments/assets/2275b674-dcfe-4b67-8bb3-f803d7797158" />

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**
No documentation update required.
<!--Add link to Talawa-Docs.-->

**Summary**
<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->
This PR updates the [AddMember.tsx](cci:7://file:///home/ash/git/gsoc/PalisadoesFoundation/talawa-admin/src/screens/OrganizationPeople/addMember/AddMember.tsx:0:0-0:0) component to include comprehensive unit tests, achieving **100% code coverage** across lines, statements, branches, and functions. 
Key changes include:
1. **Added Unit Tests:** Created [AddMember.spec.tsx](cci:7://file:///home/ash/git/gsoc/PalisadoesFoundation/talawa-admin/src/screens/OrganizationPeople/addMember/AddMember.spec.tsx:0:0-0:0) covering all functionality:
   - Success and error states for adding existing users.
   - Success and error states for creating new users.
   - Pagination logic (inc. edge cases with missing cursors).
   - Search functionality.
   - Form validation (password matching).
   - "No users found" and "Error loading users" states.
   
2. **Internationalization Fixes:** Refactored [AddMember.tsx](cci:7://file:///home/ash/git/gsoc/PalisadoesFoundation/talawa-admin/src/screens/OrganizationPeople/addMember/AddMember.tsx:0:0-0:0) to replace hardcoded strings (e.g., "Page", "No users found", "Error loading users") with translation keys used in the rest of the app. This fixed pending i18n check failures.

3. **Defensive Mocking:** Implemented robust mocks for `@mui/material/TablePagination` and `PageHeader` to expose internal events and ensuring even defensive guard clauses (like disabled buttons) could be tested for full branch coverage.

**Does this PR introduce a breaking change?**
No
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass


**Other information**
**Coverage Report:**
=============================== Coverage summary =============================== Statements : 100% ( 131/131 ) Branches : 100% ( 57/57 ) Functions : 100% ( 28/28 ) Lines : 100% ( 122/122 )
<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->
Yes

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Localization**
  * Added translations for pagination labels ("Previous Page," "Next Page," "Page number") across English, Spanish, French, Hindi, and Chinese.
  * Added translations for member management actions and event-related features across multiple languages.

* **Bug Fixes**
  * Improved error handling in member management screen with better error messages and user feedback.

* **Accessibility**
  * Updated pagination control labels with improved accessibility descriptions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes commented:
Please fix the conflicting files

ashutoshsao commented:
@coderabbitai what are the changes we need to make ?

palisadoes commented:
@coderabbitai full review including dismissed and resolved comments

palisadoes reviewed (changes requested):
Please fix. The code coverage for your patch is under the threshold. Ensure that it approaches 100%

- https://github.com/PalisadoesFoundation/talawa-admin/pull/5966#issuecomment-3703192517
        <img width="931" height="453" alt="image" src="https://github.com/user-attachments/assets/b8c413a2-41e0-4b67-960e-8bf889e1c87f" />

palisadoes commented:
@coderabbitai full review including dismissed and resolved comments

ashutoshsao commented:
> ## ⚠️ Critical Issues Found - Changes Required
> @ashutoshsao, thank you for the continued work on this PR. However, my comprehensive review has identified that **2 of the 4 critical error-handling test issues remain unresolved**.
> 
> ## Critical Issues Requiring Fixes
> ### 1. Missing Error Assertions in Two Tests ❌
> **Lines ~605-650**: `'adds an existing user to organization error'` test
> 
> * **Issue**: Test exercises the error path but does NOT verify that `NotificationToast.error` is called
> * **Required Fix**: Add this assertion before the test ends:
> 
> ```ts
> await waitFor(() => {
>   expect(NotificationToast.error).toHaveBeenCalled();
> });
> ```
> 
> **Lines ~893-940**: `'creates a new user error'` test
> 
> * **Issue**: Test exercises the error path but does NOT verify that `NotificationToast.error` is called
> * **Required Fix**: Add this assertion before the test ends:
> 
> ```ts
> await waitFor(() => {
>   expect(NotificationToast.error).toHaveBeenCalled();
> });
> ```
> 
> ## ✅ Successfully Resolved Issues
> Great work on fixing these items:
> 
> * ✅ i18n violations: All aria-labels properly internationalized
> * ✅ Mock cleanup: Changed to `vi.clearAllMocks()`
> * ✅ Test organization: All tests inside describe block
> * ✅ Test naming: Improved clarity
> * ✅ Mock buttons: Added `type="button"` attributes
> * ✅ Two other error tests (password mismatch & missing fields) now have proper assertions
> 
> ## CI/CD Status
> * All 12 test shards: ✅ SUCCESS
> * Lint/type-check: ✅ SUCCESS
> * CodeCov: ✅ SUCCESS (both patch and project)
> * E2E tests: ✅ SUCCESS
> * All other checks: ✅ SUCCESS
> 
> Once you add the missing assertions to those 2 tests, the PR will be ready for approval.
> 
> **Current HEAD commit**: `dfe0959f711f57164dbe4c92679565079fd07a32`

@coderabbitai these assertions are already present in src/screens/AdminPortal/OrganizationPeople/addMember/AddMember.spec.tsx
